### PR TITLE
Additional index settings provider validation

### PR DESCRIPTION
**Pull request by martijnvg**
Fail if an index settings provider adds a setting that was added by another index settings provider.

### CODE REVIEW

**Review by dnhatn — APPROVED** [review]
Makes sense to me. Thanks @martijnvg.

File: server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexService.java [hunk 1]
```diff
@@ -990,19 +989,20 @@ static Settings aggregateIndexSettings(
...
+                var newAdditionalSettings = provider.getAdditionalIndexSettings(
+                    request.index(),
+                    request.dataStreamName(),
+                    timeSeriesTemplate,
+                    currentState.getMetadata(),
+                    resolvedAt,
+                    templateAndRequestSettings,
+                    combinedTemplateMappings
                 );
+                validateAdditionalSettings(provider, newAdditionalSettings, additionalIndexSettings);
```
Comment (gmarouli) [inline server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexService.java:1004]: Here we validate the ` newAdditionalSettings` against the `additionalSettings` seen so far and not values previously set by the user for example `templateAndRequestSettings`. Are these 2 settings sets disjoint ?

Comment (martijnvg) [reply server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexService.java:1004]: The `newAdditionalSettings` variable contains any settings that the current index settings provider might have set. The `additionalIndexSettings` variable tracks any settings that any index settings provider may have set while looping over the list of index settings providers. The new validation requires that `newAdditionalSettings` cannot contain any settings that are already defined in  `additionalIndexSettings`.  So yes, these two Settings variables should be a disjoint.

**Review by gmarouli — APPROVED** [review]
LGTM

**Comment by martijnvg** [comment]
Thanks Mary and Nhat for reviewing!

**Pull request merged**